<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>QtQuatschEditor</class>
 <widget class="QWidget" name="QtQuatschEditor">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>797</width>
    <height>691</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QtQuatschEditor</string>
  </property>
  <property name="styleSheet">
   <string>     /*QWidget {
         background-color: qlineargradient(x1: 0, y1: 1, x2: 0, y2: 0,
                                           stop: 0 #a6a7aa, stop: 1 #dadbde);
      }*/
    
     QPushButton {
         border: 2px solid #8f8f91;
         border-radius: 6px;
         background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                           stop: 0 #f6f7fa, stop: 1 #dadbde);
         min-width: 80px;
     }
    
     QPushButton:disabled {
	color: rgb(200, 200, 200);
     }

     QPushButton:pressed {
         background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                           stop: 0 #dadbde, stop: 1 #f6f7fa);
     }

     QPushButton:flat {
         border: none; /* no border for a flat push button */
     }

     QPushButton:default {
         border-color: navy; /* make the default button prominent */
     }

/*-----------------------------------------*/
 QComboBox {
     border: 1px solid gray;
     border-radius: 3px;
     padding: 1px 18px 1px 3px;
     min-width: 6em;
 }

 QComboBox:editable {
     background: white;
 }

 QComboBox:!editable, QComboBox::drop-down:editable {
      background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                  stop: 0 #E1E1E1, stop: 0.4 #DDDDDD,
                                  stop: 0.5 #D8D8D8, stop: 1.0 #D3D3D3);
 }

 /* QComboBox gets the &quot;on&quot; state when the popup is open */
 QComboBox:!editable:on, QComboBox::drop-down:editable:on {
     background: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                 stop: 0 #D3D3D3, stop: 0.4 #D8D8D8,
                                 stop: 0.5 #DDDDDD, stop: 1.0 #E1E1E1);
 }

 QComboBox:on { /* shift the text when the popup opens */
     padding-top: 3px;
     padding-left: 4px;
 }

 QComboBox::drop-down {
     subcontrol-origin: padding;
     subcontrol-position: top right;
     width: 15px;

     border-left-width: 1px;
     border-left-color: darkgray;
     border-left-style: solid; /* just a single line */
     border-top-right-radius: 3px; /* same radius as the QComboBox */
     border-bottom-right-radius: 3px;
 }

 QComboBox::down-arrow {
     image: url(/usr/share/icons/crystalsvg/16x16/actions/1downarrow.png);
 }

 QComboBox::down-arrow:on { /* shift the arrow when popup is open */
     top: 1px;
     left: 1px;
 }</string>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedKingdom"/>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="8" rowspan="3">
    <layout class="QVBoxLayout" name="editorLayout_">
     <property name="sizeConstraint">
      <enum>QLayout::SetMaximumSize</enum>
     </property>
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>224</width>
         <height>72</height>
        </size>
       </property>
       <property name="styleSheet">
        <string>QGroupBox {
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #E0E0E0, stop: 1 #FFFFFF);
     border: 2px solid gray;
     border-radius: 5px;
     margin-top: 1ex; /* leave space at the top for the title */
 }
 QGroupBox::title {
     subcontrol-origin: margin;
     subcontrol-position: top center; /* position at the top center */
     padding: 0 3px;
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #FFOECE, stop: 1 #FFFFFF);
 }</string>
       </property>
       <property name="title">
        <string>Navigation</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item>
         <widget class="QPushButton" name="completeViewButton">
          <property name="text">
           <string>Complete View</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="centerOnSelectedButton">
          <property name="text">
           <string>Center on Selected</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="styleSheet">
        <string>QGroupBox {
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #E0E0E0, stop: 1 #FFFFFF);
     border: 2px solid gray;
     border-radius: 5px;
     margin-top: 1ex; /* leave space at the top for the title */
 }
 QGroupBox::title {
     subcontrol-origin: margin;
     subcontrol-position: top center; /* position at the top center */
     padding: 0 3px;
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #FFOECE, stop: 1 #FFFFFF);
 }


     QPushButton {
         border: 2px solid #8f8f91;
         border-radius: 6px;
         background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                           stop: 0 #f6f7fa, stop: 1 #daebde);
         min-width: 80px;
     }</string>
       </property>
       <property name="title">
        <string>Node Insertion</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_5">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QPushButton" name="insertLeftSiblingButton">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>New sibling before</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="insertRightSiblingButton">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="text">
               <string>New sibling after</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <widget class="QPushButton" name="addNodeButton">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="styleSheet">
             <string/>
            </property>
            <property name="text">
             <string>New child</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="Line" name="line">
          <property name="styleSheet">
           <string>color: black;</string>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="lineWidth">
           <number>1</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="deleteNodeButton">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="styleSheet">
           <string>         background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                           stop: 0 #f6f7fa, stop: 1 #fadbde);</string>
          </property>
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="styleSheet">
        <string>QGroupBox {
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #E0E0E0, stop: 1 #FFFFFF);
     border: 2px solid gray;
     border-radius: 5px;
     margin-top: 1ex; /* leave space at the top for the title */
 }
 QGroupBox::title {
     subcontrol-origin: margin;
     subcontrol-position: top center; /* position at the top center */
     padding: 0 3px;
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #FFOECE, stop: 1 #FFFFFF);
 }</string>
       </property>
       <property name="title">
        <string>Node Movement</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="1">
         <widget class="QPushButton" name="moveNodeUpButton">
          <property name="text">
           <string>Move up</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="asParentsSiblingButton">
          <property name="text">
           <string>As parent's sibling</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="asLeftSiblingsChildButton">
          <property name="text">
           <string>As left sibling's child</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QPushButton" name="moveNodeDownButton">
          <property name="text">
           <string>Move down</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="asRightSiblingsChildButton">
          <property name="text">
           <string>As right sibling's child</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="nodePropertiesGroupBox">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="styleSheet">
        <string>QGroupBox {
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #E0E0E0, stop: 1 #FFFFFF);
     border: 2px solid gray;
     border-radius: 5px;
     margin-top: 1ex;
 }
 QGroupBox::title {
     subcontrol-origin: margin;
     subcontrol-position: top center;
     padding: 0 3px;
     background-color: qlineargradient(x1: 0, y1: 0, x2: 0, y2: 1,
                                       stop: 0 #FFOECE, stop: 1 #FFFFFF);
 }</string>
       </property>
       <property name="title">
        <string>Node Properties</string>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
        <item>
         <widget class="QComboBox" name="typeCombo">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="frame">
           <bool>true</bool>
          </property>
          <item>
           <property name="text">
            <string>Undefined</string>
           </property>
           <property name="icon">
            <iconset resource="../resources/icons.qrc">
             <normaloff>:/aggregate/error</normaloff>:/aggregate/error</iconset>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Parameter</string>
           </property>
           <property name="icon">
            <iconset resource="../resources/icons.qrc">
             <normaloff>:/aggregate/parameter</normaloff>:/aggregate/parameter</iconset>
           </property>
          </item>
          <item>
           <property name="text">
            <string>User Constant</string>
           </property>
           <property name="icon">
            <iconset resource="../resources/icons.qrc">
             <normaloff>:/aggregate/user-constant</normaloff>:/aggregate/user-constant</iconset>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Predefined Constant</string>
           </property>
           <property name="icon">
            <iconset resource="../resources/icons.qrc">
             <normaloff>:/aggregate/predefined-constant</normaloff>:/aggregate/predefined-constant</iconset>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Addition</string>
           </property>
           <property name="icon">
            <iconset resource="../resources/icons.qrc">
             <normaloff>:/aggregate/addition</normaloff>:/aggregate/addition</iconset>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Subtraction</string>
           </property>
           <property name="icon">
            <iconset resource="../resources/icons.qrc">
             <normaloff>:/aggregate/subtraction</normaloff>:/aggregate/subtraction</iconset>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Multiplication</string>
           </property>
           <property name="icon">
            <iconset resource="../resources/icons.qrc">
             <normaloff>:/aggregate/multiplication</normaloff>:/aggregate/multiplication</iconset>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Division</string>
           </property>
           <property name="icon">
            <iconset resource="../resources/icons.qrc">
             <normaloff>:/aggregate/division</normaloff>:/aggregate/division</iconset>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Exponentiate</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Minimum</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Maximum</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Negative</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Lerp</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>And</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Or</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Not</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Less-Than (&lt;)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Less-Than-Or-Equal (&lt;=)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Greater-Than (&gt;)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Greater-Than-Or-Equal (&gt;=)</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Inverse</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Sine</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Cosine</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Floor</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Absolute</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Truncation</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Fractional</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Sqrt</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Log</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Log10</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Exp</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Noise</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>IfThenElse</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Layer-Noise</string>
           </property>
          </item>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="helpText">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="textFormat">
           <enum>Qt::RichText</enum>
          </property>
          <property name="scaledContents">
           <bool>true</bool>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QVBoxLayout" name="nodePropertiesLayout"/>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <spacer name="verticalSpacer">
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="sizeType">
        <enum>QSizePolicy::Expanding</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>20</width>
         <height>1</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_5">
       <property name="text">
        <string>PushButton</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="3" column="2">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QLabel" name="heightmapLabel">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Maximum" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="styleSheet">
        <string>QLabel {
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1,
             stop:0 rgb(108, 128, 108), 
             stop:1 rgb(118, 118, 118));
     border: 2px solid gray;
     border-radius: 5px;
     margin-top: 1ex; /* leave space at the top for the title */
 }
</string>
       </property>
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QGraphicsView" name="heightmapView">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="styleSheet">
        <string>QGraphicsView {
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1,
             stop:0 rgb(108, 128, 108), 
             stop:1 rgb(118, 118, 118));
     border: 2px solid gray;
     border-radius: 5px;
     margin-top: 1ex; /* leave space at the top for the title */
 }
</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0" rowspan="3">
    <widget class="QScrollArea" name="scrollArea_2">
     <property name="sizePolicy">
      <sizepolicy hsizetype="MinimumExpanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="maximumSize">
      <size>
       <width>172</width>
       <height>16777215</height>
      </size>
     </property>
     <property name="widgetResizable">
      <bool>true</bool>
     </property>
     <widget class="QWidget" name="scrollAreaWidgetContents_2">
      <property name="geometry">
       <rect>
        <x>0</x>
        <y>0</y>
        <width>170</width>
        <height>671</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <item>
        <widget class="QTreeWidget" name="nodeTypesTreeWidget">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Expanding">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>16777215</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="lineWidth">
          <number>1</number>
         </property>
         <property name="dragEnabled">
          <bool>true</bool>
         </property>
         <property name="dragDropMode">
          <enum>QAbstractItemView::DragOnly</enum>
         </property>
         <property name="iconSize">
          <size>
           <width>24</width>
           <height>24</height>
          </size>
         </property>
         <property name="indentation">
          <number>8</number>
         </property>
         <property name="rootIsDecorated">
          <bool>true</bool>
         </property>
         <property name="uniformRowHeights">
          <bool>false</bool>
         </property>
         <property name="animated">
          <bool>true</bool>
         </property>
         <property name="headerHidden">
          <bool>true</bool>
         </property>
         <attribute name="headerVisible">
          <bool>false</bool>
         </attribute>
         <column>
          <property name="text">
           <string>1</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item row="1" column="2" rowspan="2">
    <widget class="QGraphicsView" name="graphicsView">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
     <property name="styleSheet">
      <string>QGraphicsView {
	background-color: qlineargradient(spread:pad, x1:0, y1:0, x2:0, y2:1,
             stop:0 rgb(108, 128, 108), 
             stop:1 rgb(118, 118, 118));
     border: 2px solid gray;
     border-radius: 5px;
     margin-top: 1ex; /* leave space at the top for the title */
 }
</string>
     </property>
     <property name="verticalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="horizontalScrollBarPolicy">
      <enum>Qt::ScrollBarAsNeeded</enum>
     </property>
     <property name="renderHints">
      <set>QPainter::Antialiasing|QPainter::HighQualityAntialiasing|QPainter::SmoothPixmapTransform|QPainter::TextAntialiasing</set>
     </property>
     <property name="dragMode">
      <enum>QGraphicsView::ScrollHandDrag</enum>
     </property>
     <property name="transformationAnchor">
      <enum>QGraphicsView::AnchorUnderMouse</enum>
     </property>
     <property name="resizeAnchor">
      <enum>QGraphicsView::AnchorUnderMouse</enum>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="../resources/icons.qrc"/>
 </resources>
 <connections/>
</ui>
